sdk/monitor/monitor-opentelemetry-exporter/src/utils/metricUtils.ts

```diff
@@ -55,7 +57,23 @@ export function resourceMetricsToEnvelope(metrics: ResourceMetrics, ikey: string
           ...baseData,
         },
       },
-    };
+      }
+    } else {
```

Comment:
You could only have the name as variable and change the value of it unless there is something else changing here